tools/src/vcf2genomicsdb_init.cc

```diff
@@ -0,0 +1,993 @@
+    return OK;
+  }
+  if (partition_config.number_of_partitions < 0) {
+    partition_config.number_of_partitions = 0;
+  }
+
+  if (!partition_config.interval_list.empty()) {
+    return process_from_interval_list(partition_config.interval_list, regions, workspace, vidmap_pb,
+                                      import_config_protobuf);
+  }
+
+  int64_t number_of_partitions = partition_config.number_of_partitions;
+  int64_t max_size_of_partition = partition_config.size_of_partitions;
+  bool merge_small_contigs = partition_config.merge_small_contigs;
+
+  if (number_of_partitions) {
+    max_size_of_partition = total_length/number_of_partitions;
+  }
+  g_logger.debug("max_size_of_partition={}", max_size_of_partition);
+  int64_t tolerance = max_size_of_partition*0.25;
+  int64_t merge_small_contigs_size = 1024*1024; // 1M
+  bool partition_by_contig = !number_of_partitions && !max_size_of_partition;
+
+  int nscaffold = 0;
+  bool processing_scaffold = false;
+  int64_t scaffold_length = 0;
+
+  int64_t tiledb_column_offset = 0;
+  for (auto region: regions) {
+    std::string chromosome = region.first;
+    int64_t length = region.second;
+    int64_t processed_length = 0;
+    int64_t start, end = 0;
+
+    g_logger.debug("Region for chromosome={} length={}", chromosome, length);
+    if (partition_by_contig) {
+      max_size_of_partition = std::max(max_size_of_partition, length);
+    }
+
+    while (processed_length < length) {
+      start = processed_length;
+      end = start + ((length-processed_length)<(max_size_of_partition+tolerance)?
+                     length-processed_length:max_size_of_partition);
+
+      // Process a scaffold
+      if (merge_small_contigs && length < merge_small_contigs_size) {
+        if (!processing_scaffold) {
+          processing_scaffold = true;
+          nscaffold++;
+        }
+        scaffold_length += length;
+        if (scaffold_length > max_size_of_partition) {
+          create_partition("scaffold"+std::to_string(nscaffold), 0, scaffold_length, tiledb_column_offset,
+                         workspace, import_config_protobuf);
+          processing_scaffold = false;
+          length = scaffold_length;
+          scaffold_length = 0;
+        }
+        break;
+      } else if (processing_scaffold) { // New region that is larger than a scaffold
+        // Write out the scaffold before proccessing the new region
+        create_partition("scaffold"+std::to_string(nscaffold), 0, scaffold_length, tiledb_column_offset,
+                         workspace, import_config_protobuf);
+        processing_scaffold = false;
+        tiledb_column_offset += scaffold_length;
+      }
+
+      create_partition(chromosome, start, end, tiledb_column_offset, workspace, import_config_protobuf);        
+      processed_length += (end - start);
+    }
+    //  assert(processed_length == length);
+    if (!processing_scaffold) {
+      tiledb_column_offset += length;
+    }
+  }
```

Comment:
I think it may be possible that we exit the for-loop in `processing_scaffold` mode and still need to create a partition here.

Reply:
Good catch. Will be fixed.